### PR TITLE
PLT-6217 Fixes system message when a member is added or has joined a team.

### PR DESCRIPTION
**Pull request by letsila**
#### Summary
Fixes case 1 and 3 of [#6483](https://github.com/mattermost/mattermost-server/issues/6483). Updates system message when a member has joined a team via invite link or was added to a team.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6217

#### Checklist
- [x] Has server changes (https://github.com/mattermost/mattermost-server/pull/7752)
- [x] Has UI changes
- [x] Includes text changes and localization file updates

### CODE REVIEW

**Comment by jwilander** [comment]
@letsila could you give this a rebase? The build doesn't like it because it's too far behind master

**Comment by esethna** [comment]
Adding the awaiting PR label until the server changes are in

**Comment by letsila** [comment]
Hi @esethna. Actually I have forgotten this on the server side PR, however that PR was merged. So how should I proceed? Should I add more commits on the merged branch or create a completely new one?

**Comment by esethna** [comment]
Please create a new PR and link it here, thanks!

**Comment by letsila** [comment]
Awaiting for the new PR to be merged.

**Comment by esethna** [comment]
Thanks!

**Comment by letsila** [comment]
It's a pleasure!

File: components/post_view/post_message_view/system_message_helpers.jsx [hunk 1]
```diff
@@ -63,13 +75,29 @@ function renderAddToChannelMessage(post, options) {
...
+}
+
 function renderRemoveFromChannelMessage(post, options) {
     const removedUsername = renderUsername(post.props.removedUsername, options);
 
     return (
         <FormattedMessage
             id='api.channel.remove_member.removed'
-            defaultMessage='{removedUsername} was removed from the channel'
+            defaultMessage='{removedUsername} was removed from the team'
```
Comment (grundleborg) [inline components/post_view/post_message_view/system_message_helpers.jsx]: This should still say `channel` when the user is just being removed from a random channel. It should only say `team` when they are actually removed from the entire team (i.e. in `town-square`).

**Comment by grundleborg** [comment]
@letsila thanks for updating the PR - I'm putting this back to PM review just so @esethna can double check the latest change.

**Comment by jasonblais** [comment]
@letsila Huge thanks for your help here

I found one small bug -- if a member is removed from a channel, the system message reads that they're removed from the team.

Repro:
1) Create a channel
2) Add a member
3) Remove that member

Observed: System message `test2 was removed from the team`
Expected: System message `test2 was removed from the channel`

image

**Comment by hmhealey** [comment]
Hi @letsila, I just wanted to let you know that I fixed that line so we can get this added in for 4.6

**Comment by letsila** [comment]
Hi @hmhealey, thanks for that. However when a user is removed from the team the message would get wrong, because it will display `{removedUsername} was removed from the channel`. There are some fixes that I should have done with my previous server-side PR. Actually, I did not now at the first place how to remove a user from a channel via the UI so I struggled a bit about how to test it. But I am ready to fix the server-side code.

**Comment by jasonblais** [comment]
@letsila Can you try rebasing again? Looks like there are additional changes in the PR

**Pull request merged**